core/modules/system/system.admin.inc

```diff
@@ -1450,6 +1450,7 @@ function system_site_information_settings_submit($form, &$form_state) {
     ->set('site_404', $form_state['values']['site_404'])
     ->save();
   backdrop_set_message(t('The configuration options have been saved.'));
+  backdrop_flush_all_caches();
```

Comment:
If moving the logic into `node_page_default()` this becomes no longer necessary.